### PR TITLE
fix bugs

### PR DESCRIPTION
**Pull request by chenzongxiong**

### CODE REVIEW

File: setup.ts [hunk 1]
```diff
@@ -161,45 +165,44 @@ class Chain {
...
-        args: string | AccountId | Uint8Array,
+        args: any,
         value = DEFAULT_DEMOCRACY_PROPOSAL_FEE
     ) {
-        await this.connect();
-        // const toAddRegistrar = await self.identityAddRegistrar(registrarAccount);
-        const result = await func(args);
+        await self.connect();
+        // @ts-ignore
+        const result = await func(...args);
```
Comment (chenzongxiong) [inline setup.ts:175]: We need to unpack function parameters. 
`func` is a function pointer, and `args` is an array of function parameters.
Suppose the declaration of `func` is `func(a: Type1, b: Type2)`. 
if we invoke `func` by calling `func(args)`. It implies the declaration of `func(Array arr)` , and `args = [a: Type1, b: Type2]`. The method will perform in a totally different way.


That's why we need to unpack the array

File: setup.ts [hunk 2]
```diff
@@ -279,48 +283,46 @@ class Chain {
...
-        let publicProps = await this.democracyPublicProps();
+        let publicProps = await self.democracyPublicProps();
         await sleep(DEFAULT_SLEEP_INTERVAL);
         if (`${publicProps.length}` === '0') {
-            // TODO_CHECK this code isn't being hit, and it's throwing build errors (remove the 2 ts-ignores to highlight the issues)
+            // TODO_CHECK self code isn't being hit, and it's throwing build errors (remove the 2 ts-ignores to highlight the issues)
             // @ts-ignore
-            await this.democracyNotePreimage(this.alice, this.identityAddRegistrar, [this.alice, registrarAccount]);
+            await self.democracyNotePreimage(self.alice, self.identityAddRegistrar, [account2registrar[`${registrarAccount.address}`]]);
             await sleep(DEFAULT_SLEEP_INTERVAL);
```
Comment (bdmason) [inline setup.ts:292]: is this because the wrong account was proposing the regisitrar?

**Review by bdmason — APPROVED** [review]
Was it just the type with `...args` and the address passed here to the last parameter here`await self.democracyNotePreimage(self.alice, self.identityAddRegistrar, [account2registrar[`${registrarAccount.address}`]]);`?

File: setup.ts (hunk 2 above)
Comment (chenzongxiong) [reply setup.ts:292]: It's also a reason.

Comment (chenzongxiong) [reply setup.ts:292]: If we simply call `await self.democracyNotePreimage(self.alice, self.identityAddRegistrar account2registrar[`${registrarAccount.address}`]); `, then your source code can work.

**Pull request merged**